File: solutions/typescript/0064.ts
function minPathSum(grid: number[][]): number {
    if (!grid || grid.length === 0 || grid[0].length === 0) {
        return 0;
    }
    const m = grid.length;
    const n = grid[0].length;
    const dp: number[] = new Array(n).fill(0);
    dp[0] = grid[0][0];
    for (let j = 1; j < n; j++) {
        dp[j] = dp[j - 1] + grid[0][j];
    }
    for (let i = 1; i < m; i++) {
        dp[0] += grid[i][0];
        for (let j = 1; j < n; j++) {
            dp[j] = Math.min(dp[j], dp[j - 1]) + grid[i][j];
        }
    }
    return dp[n - 1];
}

function 最小路径和(...args: any[]): any {
    if (args.length >= 1 && Array.isArray(args[0])) {
        return minPathSum(args[0]);
    }
    return 0;
}

export default minPathSum;

File: solutions/rust/0064.rs
/*
 * 最小路径和（LeetCode 64）
 * m×n 网格，从左上角只能向右或向下走，到右下角的最小路径和。
 * DP 一维：dp[j] = min(dp[j], dp[j-1]) + grid[i][j]。时间 O(mn)，空间 O(n)。
 */
impl Solution {
    pub fn min_path_sum(grid: Vec<Vec<i32>>) -> i32 {
        if grid.is_empty() || grid[0].is_empty() {
            return 0;
        }
        let m = grid.len();
        let n = grid[0].len();
        let mut dp = vec![0; n];
        dp[0] = grid[0][0];
        for j in 1..n {
            dp[j] = dp[j - 1] + grid[0][j];
        }
        for i in 1..m {
            dp[0] += grid[i][0];
            for j in 1..n {
                dp[j] = dp[j].min(dp[j - 1]) + grid[i][j];
            }
        }
        dp[n - 1]
    }

    pub fn 最小路径和(nums: Vec<i32>, target: i32) -> Vec<i32> {
        // 占位实现（根据实际需要调整参数）
        vec![]
    }
}